src/transformers/generation/tf_utils.py

```diff
@@ -1430,14 +1430,22 @@ def _get_logits_warper(
         # instantiate warpers list
         warpers = TFLogitsProcessorList()
 
-        # the following idea is largely copied from this PR: https://github.com/huggingface/transformers/pull/5420/files
-        # all samplers can be found in `generation_utils_samplers.py`
+        # In beam methods, we need to keep at least one non-eos token to explore continuations that might have a
+        # better score (i.e. keep len(generation_config.eos_token_id) + 1)
+        if generation_config.num_beams > 1:
+            if isinstance(generation_config.eos_token_id, list):
+                min_tokens_to_keep = len(generation_config.eos_token_id) + 1
+            else:
+                min_tokens_to_keep = 2
```

Comment:
```suggestion
            min_tokens_to_keep = len(list(generation_config.eos_token_id)) + 1
```

Reply:
The catch is that `generation_config.eos_token_id` can be `None`, and `len(list(None))` throws `TypeError: 'NoneType' object is not iterable` :(

Reply:
Ah right 💯